extensions/metube/src/add-to-metube.tsx

```diff
@@ -0,0 +1,79 @@
+import { Form, ActionPanel, Action, showToast, getPreferenceValues, Toast } from "@raycast/api";
+import { addToQueue } from "./helper.js";
+import { showFailureToast } from "@raycast/utils";
+
+type Values = {
+  url: string;
+  quality: string;
+  format: string;
+};
+
+interface Preferences {
+  metubeUrl: string;
+  quality: string;
+  format: string;
+}
+
+export default function Command() {
+  const preferences = getPreferenceValues<Preferences>();
+  const quality = preferences.quality;
+  const format = preferences.format;
+
+  async function handleSubmit(values: Values) {
+    if (!values.url) {
```

Comment:
You can use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) which has validation, errors and handles states

Reply:
Oh! I see, thank you, I was not aware of this, looks interesting, I will implement this for sure.